### PR TITLE
 "How I survived Git" tips 

### PR DESCRIPTION
**Pull request by matklad**
bors r+
🤖

### CODE REVIEW

File: xtask/tests/tidy.rs [hunk 1]
```diff
@@ -52,15 +52,36 @@ fn rust_files_are_tidy() {
...
+
+  # Abort in-progress rebase, if any.
+  $ git rebase --abort
+
+  # Make the branch point to the latest commit from master,
+  # while maintaining your local changes uncommited.
+  $ git reset --soft origin/master
+
+  # Commit all changes in a single batch.
+  $ git commit -am'My changes'
```
Comment (bjorn3) [inline xtask/tests/tidy.rs:74]:
```suggestion
  $ git commit -am 'My changes'
```

Comment (matklad) [reply xtask/tests/tidy.rs:74]: Hm, I personally never type a space here, and it works

Comment (bjorn3) [reply xtask/tests/tidy.rs:74]: Surprising. `-am'My changes'` and `'-amMy changes'` are passed the same on Unix. Most programs would interpret it as `-a -m -M -y '- ' -c -h -a -n -g -e -s`. There seems to be a special case for `-m` in `git commit`.

Comment (matklad) [reply xtask/tests/tidy.rs:74]: Wow!

I wonder if might be the behavior of my shell (fish)?

Comment (bjorn3) [reply xtask/tests/tidy.rs:74]: It also works fine with my shell (zsh)

**Pull request merged**